pluggy.py

```diff
@@ -605,10 +623,10 @@ def varnames(func):
 
     assert "self" not in args  # best naming practises check?
     try:
-        cache["_varnames"] = args
+        cache["_varnames"] = args, defaults
     except TypeError:
```

Comment:
@RonnyPfannschmidt huh that's an interesting point.

I've always understood that function lifetimes last from module import until interpreter exit unless you explicitly delete references to them. Also consider `varnames()` is only normally called once at `PluginManager.register()` time (really only once for each `hookspec` or `hookimpl`) so I might even go so far as to say this whole cache idea is a pre-mature optimization.

That being said @hpk42 had it there originally and I see no strong reason to remove it.